src/resolve.rs

```diff
@@ -0,0 +1,15 @@
+use std::net::SocketAddr;
+
+pub trait Resolver: std::fmt::Debug + Send + Sync {
+    fn resolve(&self, netloc: &str) -> std::io::Result<Box<dyn Iterator<Item = SocketAddr>>>;
+}
+
+#[derive(Debug)]
+pub(crate) struct StdResolver;
+
+impl Resolver for StdResolver {
+    fn resolve(&self, netloc: &str) -> std::io::Result<Box<dyn Iterator<Item = SocketAddr>>> {
```

Comment:
Why return an iterator when `ureq` always takes the first item? Also why is the Box required?

Reply:
`ureq` doesn't always take the first item. There is a succession of fallbacks in https://github.com/algesten/ureq/blob/6a88c2c8bfbb0bac97029265a42703a17c76e01c/src/stream.rs#L408.

An alternative to the `Box<dyn Iterator>` would be returning a `Vec<SocketAddr>` directly. If aiming for just simpler resolvers, `Vec<SocketAddr>` might be simpler.